src/platform/qt/library/LibraryController.cpp

```diff
@@ -55,14 +58,52 @@ LibraryController::LibraryController(QWidget* parent, const QString& path, Confi
 
 	mLibraryAttachGameDB(m_library.get(), GBAApp::app()->gameDB());
 
-	m_libraryTree = std::make_unique<LibraryTree>(this);
-	addWidget(m_libraryTree->widget());
-
-	m_libraryGrid = std::make_unique<LibraryGrid>(this);
-	addWidget(m_libraryGrid->widget());
+	m_libraryModel = new LibraryModel(this);
+
+	addWidget(m_treeView = new QTreeView(this));
+	m_treeModel = new QSortFilterProxyModel(this);
+	m_treeModel->setSourceModel(m_libraryModel);
+	m_treeModel->setSortRole(Qt::EditRole);
+	m_treeView->setModel(m_treeModel);
+	m_treeView->setSortingEnabled(true);
+	m_treeView->setAlternatingRowColors(true);
+
+	addWidget(m_listView = new QListView(this));
+	m_listModel = new QSortFilterProxyModel(this);
+	m_listModel->setSourceModel(m_libraryModel);
+	m_listModel->setSortRole(Qt::EditRole);
+	m_listView->setModel(m_listModel);
+
+	QObject::connect(m_treeView, &QAbstractItemView::activated, this, &LibraryController::startGame);
+	QObject::connect(m_listView, &QAbstractItemView::activated, this, &LibraryController::startGame);
+	QObject::connect(m_treeView->header(), &QHeaderView::sortIndicatorChanged, this, &LibraryController::sortChanged);
+
+	QTimer* expandThrottle = new QTimer(this);
```

Comment:
This is...sort of a weird way to do this. I suppose it works, but...

Anyway, in general I'd make something like this a member variable, even if I don't access it after this function, just so that it lives in the same allocation. Not sure that makes sense in the case of d-pointer classes though.